src/auth/token-verifier.ts

```diff
@@ -237,13 +247,15 @@ export class FirebaseTokenVerifier {
    * @return {Promise<DecodedIdToken>} A promise that resolves with the decoded JWT claims on successful
    *     verification.
    */
-  private verifyJwtSignatureWithKey(jwtToken: string, publicKey: string): Promise<DecodedIdToken> {
+  private verifyJwtSignatureWithKey(jwtToken: string, publicKey: string | null): Promise<DecodedIdToken> {
     const verifyJwtTokenDocsMessage = ` See ${this.tokenInfo.url} ` +
       `for details on how to retrieve ${this.shortNameArticle} ${this.tokenInfo.shortName}.`;
     return new Promise((resolve, reject) => {
-      jwt.verify(jwtToken, publicKey, {
+      // The types are wrong here so it won't accept 'null' for publicKey, but testing (including our unit tests)
+      // confirms that this works and is the required value when validating a token with alg: 'none'
+      jwt.verify(jwtToken, publicKey as any, {
```

Comment:
Drive by comment: consider using `jwt.decode` for emulator only, which has the right typings and explicitly skips signature verification. We do this in Auth Emulator.

Reply:
Yeah I considered that my reasoning for this was that we use `verify` in the actual code and I want the tests to prove that won't throw.

Reply:
I submitted a PR to fix the types:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48348

If it gets merged I can remove the hack.

Reply:
nit: Have you tried `""` (empty string) instead of `null`?

Reply:
Wow I think that worked!